### PR TITLE
Cookie Encryption

### PR DESCRIPTION
**Pull request by HB9HIL**
save encrypting cookies to keep me logged in

### CODE REVIEW

**Review by int2001 — CHANGES_REQUESTED** [review]
It's working. Well done.

Some minor (but important) things, as discussed:
- Disable Login via remember_me Cookie, if secret wasn't altered.
- pse set httpOnly as well at the cookie-options.

File: application/controllers/User.php [hunk 1]
```diff
@@ -798,7 +868,23 @@ function login() {
...
+				// Create a keep_login cookie
+				if ($this->input->post('keep_login') == '1') {
+
+					$encrypted_string = $this->user_model->keep_cookie_hash($data['user']->user_id);
+
+					$cookie = array(
+						'name'   => 'keep_login',
+						'value'  => $encrypted_string,
+						'expire' => '2592000',  // 30 days
+						'secure' => TRUE
```
Comment (int2001) [inline application/controllers/User.php:881]: This one will perhaps fail, if WL runs in a non-TLS-Environment.

Comment (HB9HIL) [reply application/controllers/User.php:884]: not sure. the 'keep logged in' checkbox is hidden if the page is called without https

Comment (int2001) [reply application/controllers/User.php:884]: fine for me as well. So no "remember me" for HTTP-only users :)

**Review by int2001 — APPROVED** [review]
Well done. Can't spot any problems.

tbh: a seconds pair of eyes on this won't hurt. it's a very sensitive area of code...

**Review by phl0 — APPROVED** [review]
Works as expected.

**Pull request merged**